### PR TITLE
Copy and rename compiled script directly for deprecated version

### LINKED ISSUE #9325: Error compiling extensions/amp-sticky-ad/1.0/amp-sticky-ad.js
Build flaky:
```
Error compiling extensions/amp-sticky-ad/1.0/amp-sticky-ad.js
ENOENT: no such file or directory, open 'build/cc/extensions_amp-sticky-ad_1.0_amp-sticky-ad.js'
```

### PR DESCRIPTION
**Pull request by zhouyx**
Should fix #9325 
have #9642 as a follow up

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -136,7 +136,8 @@ declareExtension('amp-viewer-integration', '0.1', {
 });
 declareExtension('amp-video', '0.1', false);
 declareExtension('amp-youtube', '0.1', false);
-
+declareAliasExtensionVersion(
```
Comment (lannka) [inline gulpfile.js]: declareExtensionVersionAlias

File: gulpfile.js [hunk 2]
```diff
@@ -44,6 +44,7 @@ var hostname3p = argv.hostname3p || '3p.ampproject.net';
 
 // All declared extensions.
 var extensions = {};
+var aliasExtensionsCopyPath = {};
```
Comment (lannka) [inline gulpfile.js]: extensionAliasFilePath

File: gulpfile.js [hunk 3]
```diff
@@ -355,6 +371,15 @@ function copyCss() {
 }
 
 /**
+ * Copy built extension to alias extension
+ */
+function copyAliasExtensions() {
```
Comment (lannka) [inline gulpfile.js]: nit: move it down to right below its usage

**Comment by zhouyx** [comment]
Hmmm travis passed, but I haven't changed https://github.com/ampproject/amphtml/blob/master/extensions/amp-sticky-ad/0.1/test/validator-amp-sticky-ad.out#L2
where it points to `amp-sticky-ad/0.1/test/validator-amp-sticky-ad.html` which doesn't exist anymore. 
@Gregable How should I fix this?

**Comment by zhouyx** [comment]
think I find it https://github.com/ampproject/amphtml/blob/master/validator/engine/validator_test.js#L88

**Comment by lannka** [comment]
to not block this PR (this PR is making travis build very flaky), we can do it separately. 
seems a wildcard `${extension}/*/test/` is needed. @Gregable thoughts?

**Comment by zhouyx** [comment]
I think we should be able to unblock the PR as long as we confirm that `validator-amp-sticky-ad.protoascii` gets applied in folder 1.0 unlike the test files.
I can revert part of the change and keep 0.1 folder as a workaround? WDYT

File: gulpfile.js [hunk 4]
```diff
@@ -165,6 +166,21 @@ function declareExtension(name, version, hasCssOrOptions, opt_noTypeCheck,
 }
 
 /**
+ * @param {string} name
+ * @param {string} version E.g. 0.1
```
Comment (lannka) [inline gulpfile.js]: can we make it an array?

**Comment by zhouyx** [comment]
....Can't just leave 0.1 folder there without build. `dep-check` failed because the function `getEntryModule` by simply set version as 0.1 ....
We really need to remove all hardcoded 0.1 path.  I created #9642.
In the meanwhile, I can't think of a good way to unblock this PR without fixing #9642 first. Best I can do is to put a `isAmpStickyAd` check to dep-check.... 😢

**Pull request merged**